### ISSUE TITLE
BigQueryConfig is not respected (caching and priority)

### ISSUE THREAD

**Issue by bbhoss** [spotify/scio#1718]
I have tried everything under the sun to get the BigQuery functionality to respect my wishes as documented, which is to set
```
  System.setProperty("bigquery.cache.enabled", "false")
  System.setProperty("bigquery.priority", "INTERACTIVE")
```

BigQueryConfig seems to think that it's interactive when interrogated:
```
scala> com.spotify.scio.bigquery.client.BigQueryConfig.priority.map(_ == "INTERACTIVE")
res2: Option[Boolean] = Some(true)
```
Yet, jobs are still scheduled as batch.

**Comment by ClaireMcGinty** [comment]
hi @bbhoss . Can you share some more information about the kind of BQ job you're writing (code snippet would be great) and how you're running it?

**Comment by bbhoss** [comment]
Thanks @ClaireMcGinty. Here's a small snippet of the code
```scala
        val patients = sc
          .typedBigQuery[Member](
            s"SELECT * from `${c.productionProject}.gold_claims.Member` WHERE identifier.patientId IS NOT NULL")
          .flatMap(member => member.MEMBER_ID)
```

`Member` is a case class that's been enhanced with typed bigquery abilities using `@BigQueryType.toTable`. Doesn't seem too exotic, so I'm wondering if it's some weird deps issue. I am happy to dig in further to debug, just not sure where to start.

**Comment by bbhoss** [comment]
Also, I will add that this happens with DirectRunner and DataflowRunner

**Comment by ClaireMcGinty** [comment]
Thanks! I haven't tested this but my hypothesis is that we need to set [.withQueryPriority(...)](https://github.com/apache/beam/blob/f6fdeaa1a65b690f1c876f77802279a23096f4c2/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L1016) on the [TypedRead](https://github.com/spotify/scio/blob/68d22b04a0d1dff4e621f5204851ce1d3f573ef3/scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala#L75) transform in `BigQuerySelect`. Would you be interested in trying this out/submitting a pull request? Otherwise we can add it to our backlog :)

**Comment by bbhoss** [comment]
I will certainly try that if I can remember how to run snapshots with our
project

**Comment by bbhoss** [comment]
So I got to the bottom of this. It turns out I had enabled forking in the project, but was setting the sysprops in sbt's initialize callback. When it forks the properties aren't copied over, but console does work, so that's why it seemed like it should work. Additionally, I'm not sure if the docs have been updated since I last tried to fix this, but I was not uppercasing my INTERACTIVE/BATCH. IMO, this is kinda a surprising thing to be picky about, and I'm happy to send a simple improvement for making it case insensitive if y'all are interested.
